crates/rspack_binding_values/src/clean_options.rs

```diff
@@ -0,0 +1,76 @@
+use napi::bindgen_prelude::{FromNapiValue, ToNapiValue};
+use napi::Either;
+use napi_derive::napi;
+use rspack_core::CleanOptions;
+use rspack_napi::napi;
+
+/// File clean options
+///
+/// A file filter is an option whether the file should be kept after clean up
+///
+/// TS Type:
+///
+/// ```typescript
+/// // in the future, we should support the following types, just like webpack
+/// // type CleanOptions = boolean | { dry?: boolean, keep?: RegExp | string | ((filename: string) => boolean) }
+///
+/// type CleanOptions = boolean | { keep?: string }
+/// ```
+#[derive(Debug)]
+pub struct JsCleanOptions(Either<bool, JsCleanFilter>);
```

Comment:
Clean options should only be sent from javascript to rust, so just add `#[napi(object, object_to_js = false)]` to it. ToNapiValue trait it not necessary. And I think the conversion logic of the clean options can be handled on the js side, and rust can just receive the converted object. Then `unsafe` can be removed

Reply:
+1. Does not look like we have to add the `unsafe` trait here.